air/src/execution_step/boxed_value/stream.rs

```diff
@@ -182,6 +202,26 @@ impl Stream {
 
         Some(iter)
     }
+
+    /// Removes empty generations updating data and returns final generation count.
+    pub(crate) fn compactify(mut self, trace_ctx: &mut TraceHandler) -> ExecutionResult<usize> {
```

Comment:
Why not `&mut self`? To make sure it is the last call to the structure?

Reply:
Yep, to provide additional hardening